src/internal/digitalAsset.ts

```diff
@@ -305,48 +351,67 @@ export async function mintDigitalAssetTransaction(args: {
     propertyValues,
   } = args;
   const convertedPropertyType = propertyTypes?.map((type) => PropertyTypeMap[type]);
-  const transaction = await generateTransaction({
+  return generateTransaction({
     aptosConfig,
     sender: creator.accountAddress,
     data: {
       function: "0x4::aptos_token::mint",
       functionArguments: [
-        collection,
-        description,
-        name,
-        uri,
-        propertyKeys ?? [],
-        convertedPropertyType ?? [],
+        new MoveString(collection),
+        new MoveString(description),
+        new MoveString(name),
+        new MoveString(uri),
+        MoveVector.MoveString(propertyKeys ?? []),
+        MoveVector.MoveString(convertedPropertyType ?? []),
         getPropertyValueRaw(propertyValues ?? [], convertedPropertyType ?? []),
       ],
     },
     options,
+    abi: mintDigitalAssetAbi,
   });
-  return transaction;
 }
 
+const transferDigitalAssetAbi: EntryFunctionABI = {
+  typeParameters: [{ constraints: [MoveAbility.KEY] }],
```

Comment:
interesting.. why do we need to define the key ability in the abi?

Reply:
So, technically, this comes directly from the API.

The type is constrained by ability, but it's not currently used in the type checking in typescript (though should be in the future).  The type needs to be able to be used as a resource lookup.

E.g. a struct with `key` but not u64, etc.